pyomo/contrib/mindtpy/util.py

```diff
@@ -565,6 +565,9 @@ def set_solver_options(opt, timing, config, solver_type, regularization=False):
     elif solver_name in {'ipopt', 'appsi_ipopt'}:
         opt.options['max_cpu_time'] = remaining
```

Comment:
Should this not be `float` as well?

Reply:
Nope. `ipopt` and `appsi_ipopt` work well with `int` `max_cpu_time`. Only `cyipopt` requires `max_cpu_time` to be `float` .